File: extra5/app/src/main/java/com/example/extra5/NewsAdapter.java
package com.example.extra5;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.extra5.db.NewsEntity;

import java.util.ArrayList;
import java.util.List;

public class NewsAdapter extends RecyclerView.Adapter<NewsAdapter.VH> {
    private final List<NewsEntity> items = new ArrayList<>();

    public void submit(List<NewsEntity> list) {
        items.clear();
        items.addAll(list);
        notifyDataSetChanged();
    }

    static class VH extends RecyclerView.ViewHolder {
        TextView title, desc;
        ImageView image;
        VH(@NonNull View itemView) {
            super(itemView);
            title = itemView.findViewById(R.id.newsTitle);
            desc = itemView.findViewById(R.id.newsDesc);
            image = itemView.findViewById(R.id.newsImage);
        }
    }

    @NonNull @Override public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new VH(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_news, parent, false));
    }
    @Override public int getItemCount() { return items.size(); }

    @Override public void onBindViewHolder(@NonNull VH holder, int position) {
        NewsEntity item = items.get(position);
        holder.title.setText(item.title);
        holder.desc.setText(item.description);
        Glide.with(holder.itemView).load(item.imageUrl).centerCrop().into(holder.image);
    }
}

File: extra5/app/src/main/java/com/example/extra5/NewsAdapter.kt
package com.example.extra5

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.extra5.db.NewsEntity

class NewsAdapter : RecyclerView.Adapter<NewsAdapter.VH>() {
    private val items = mutableListOf<NewsEntity>()
    fun submit(list: List<NewsEntity>) { items.clear(); items.addAll(list); notifyDataSetChanged() }
    class VH(parent: ViewGroup): RecyclerView.ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_news, parent, false)) {
        val title: TextView = itemView.findViewById(R.id.newsTitle)
        val desc: TextView = itemView.findViewById(R.id.newsDesc)
        val image: ImageView = itemView.findViewById(R.id.newsImage)
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = VH(parent)
    override fun getItemCount() = items.size
    override fun onBindViewHolder(holder: VH, position: Int) {
        val item = items[position]
        holder.title.text = item.title
        holder.desc.text = item.description
        Glide.with(holder.itemView).load(item.imageUrl).centerCrop().into(holder.image)
    }
}
